core/src/main/java/ai/timefold/solver/core/impl/domain/solution/cloner/gizmo/GizmoSolutionClonerImplementor.java

```diff
@@ -137,51 +160,49 @@ public static void defineClonerFor(Supplier<GizmoSolutionClonerImplementor> impl
             Set<Class<?>> solutionClassSet,
             Map<Class<?>, GizmoSolutionOrEntityDescriptor> memoizedSolutionOrEntityDescriptorMap,
             Set<Class<?>> deepClonedClassSet) {
-        GizmoSolutionClonerImplementor implementor = implementorSupplier.get();
+        var implementor = implementorSupplier.get();
         // Classes that are not instances of any other class in the collection
         // have a subclass level of 0.
         // Other classes subclass level is the maximum of the subclass level
         // of the classes it is a subclass of + 1
-        Set<Class<?>> deepCloneClassesThatAreNotSolutionSet =
+        var deepCloneClassesThatAreNotSolutionSet =
                 deepClonedClassSet.stream()
                         .filter(clazz -> !solutionClassSet.contains(clazz) && !clazz.isArray())
                         .filter(GizmoSolutionClonerImplementor::isCloneableClass)
                         .collect(Collectors.toSet());
 
-        Comparator<Class<?>> instanceOfComparator = getInstanceOfComparator(deepClonedClassSet);
-        SortedSet<Class<?>> deepCloneClassesThatAreNotSolutionSortedSet = new TreeSet<>(instanceOfComparator);
+        var instanceOfComparator = getInstanceOfComparator(deepClonedClassSet);
+        var deepCloneClassesThatAreNotSolutionSortedSet = new TreeSet<>(instanceOfComparator);
         deepCloneClassesThatAreNotSolutionSortedSet.addAll(deepCloneClassesThatAreNotSolutionSet);
 
-        implementor.createFields(classCreator);
-        implementor.createConstructor(classCreator);
-        implementor.createSetSolutionDescriptor(classCreator, solutionDescriptor);
-        implementor.createCloneSolution(classCreator, solutionDescriptor);
-        implementor.createCloneSolutionRun(classCreator, solutionDescriptor, solutionClassSet,
-                memoizedSolutionOrEntityDescriptorMap,
-                deepCloneClassesThatAreNotSolutionSortedSet, instanceOfComparator);
-
-        for (Class<?> deepClonedClass : deepCloneClassesThatAreNotSolutionSortedSet) {
-            implementor.createDeepCloneHelperMethod(classCreator, deepClonedClass, solutionDescriptor,
-                    memoizedSolutionOrEntityDescriptorMap,
-                    deepCloneClassesThatAreNotSolutionSortedSet);
+        var clonerDescriptor = new ClonerDescriptor(solutionDescriptor, memoizedSolutionOrEntityDescriptorMap,
+                deepCloneClassesThatAreNotSolutionSortedSet,
+                classCreator);
+
+        implementor.createFields(clonerDescriptor);
+        implementor.createConstructor(clonerDescriptor);
+        implementor.createSetSolutionDescriptor(clonerDescriptor);
+        implementor.createCloneSolution(clonerDescriptor);
+        implementor.createCloneSolutionRun(clonerDescriptor, solutionClassSet, instanceOfComparator);
```

Comment:
Is there ever a situation where these methods would not be called together?
If not, I suggest they be merged in a single method. (Under the hood, they can still be separate, just not accessible to the public; I'd like to avoid situation where the object is incompletely initialized, if it can be avoided.)

Reply:
This does not initialize an object; it literally creating the fields/constructors/methods on the class object. I prefer to have them separate, since Quarkus overrides them and separation of responsibilities.